packages/DropdownMenu/src/DropdownMenu.js

```diff
@@ -0,0 +1,111 @@
+import React from "react";
+import PropTypes from "prop-types";
+import Popover from "@paprika/popover";
+import ContentContainerStyled from "./ContentContainer.styles";
+import Confirmation from "./Confirmation/Confirmation";
+import Divider from "./Divider/Divider";
+import Trigger from "./Trigger/Trigger";
+import Item from "./Item/Item";
+
+const { alignTypes, oneOf, node, string, func } = PropTypes;
+
+const propTypes = {
+  /** Alignment of the dropdown menu */
+  align: oneOf(alignTypes),
+
+  /** Children should consist of <Dropdown.Item /> */
+  children: node.isRequired,
+
+  /** Optional custom classes */
+  className: string,
+
+  /** Render prop for rendering the trigger element that toggles the dropdown */
+  renderTrigger: func.isRequired,
+};
+
+const defaultProps = {
+  align: "bottom",
+  className: "",
+};
+
+const DropdownMenu = props => {
+  const { align, children, className } = props;
+  const [isOpen, setIsOpen] = React.useState(false);
+  const [isConfirming, setIsConfirming] = React.useState(false);
+  const [renderConfirmation, setRenderConfirmation] = React.useState(null);
+
+  const handleCloseMenu = () => {
+    setIsOpen(false);
+    setIsConfirming(false);
+    setRenderConfirmation(null);
+  };
+
+  const handleOpenMenu = () => {
+    setIsOpen(true);
+  };
+
+  const getTriggerStateAndHelpers = () => {
+    return {
+      isOpen,
+      handleOpenMenu,
+    };
+  };
+
+  const handleShowConfirmation = renderConfirmation => () => {
+    setIsConfirming(prevIsConfirmingState => !prevIsConfirmingState);
+    setRenderConfirmation(prevIsConfirmingState => (prevIsConfirmingState ? null : renderConfirmation));
+  };
+
+  const renderContent = () => {
+    if (isConfirming) {
+      return renderConfirmation(handleCloseMenu);
+    }
+
+    return (
+      <ContentContainerStyled isOpen={isOpen}>
+        {React.Children.toArray(children).map(child => {
```

Comment:
Do we need `toArray()` here, or can we just use `React.Children.map()`?

Reply:
Well we arn't sure what is being passed in a children in this the items could in theory contain more than just text, so I'm not sure if we have a good solution for setting the key as we don't allow to use `index` Ideas?

Reply:
I mean use doing this instead:
`React.Children.map(children, child => { ... });`
I saves a step (calling `toArray()`) and should be a bit more performant:
https://stackoverflow.com/questions/44721768/react-children-map-vs-react-children-toarray-map

Reply:
@tristanjasper Have you addressed this?

Reply:
What I'm thinking is, we should be able to avoid using `cloneElement` here as well similarly to https://github.com/acl-services/paprika/pull/134/commits/f5063084fb5c90ab0477b913fc0ac8bd7b759b04.
We can put the two functions onto context instead (unconditionally). This would make the component more composable as it wouldn't rely on `DropdownMenu.Item` being its direct child.